### PR TITLE
Add support for OCI package type #386

### PR DESCRIPTION
**Pull request by jcoste-orange**
- [x] All [tests](https://github.com/jfrog/artifactory-client-java#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
-----

### CODE REVIEW

**Comment by jcoste-orange** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jcoste-orange** [comment]
I don't understand why some tests are failling as they are not related to the pull request.

Did someone has an idea ?

File: services/src/main/java/org/jfrog/artifactory/client/model/repository/settings/impl/OciRepositorySettingsImpl.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.jfrog.artifactory.client.model.PackageType;
+import org.jfrog.artifactory.client.model.impl.PackageTypeImpl;
+import org.jfrog.artifactory.client.model.repository.settings.OciRepositorySettings;
+
+public class OciRepositorySettingsImpl extends DockerRepositorySettingsImpl implements OciRepositorySettings {
+    public static String defaultLayout = "simple-default";
+    
+    public OciRepositorySettingsImpl() {
+        super();
+    }
```
Comment (yahavi) [inline services/src/main/java/org/jfrog/artifactory/client/model/repository/settings/impl/OciRepositorySettingsImpl.java:12]: Looks unnecessary 
```suggestion
```

File: services/src/main/java/org/jfrog/artifactory/client/model/repository/settings/impl/OciRepositorySettingsImpl.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof OciRepositorySettingsImpl)) return false;
+        if (!super.equals(o)) return false;
+
+        OciRepositorySettingsImpl that = (OciRepositorySettingsImpl) o;
+
+        return true;
```
Comment (yahavi) [inline services/src/main/java/org/jfrog/artifactory/client/model/repository/settings/impl/OciRepositorySettingsImpl.java:26]: Please consider this change:
```suggestion
        if (!(o instanceof OciRepositorySettingsImpl)) {
            return false;
        }
        return super.equals(o);
```

File: services/src/main/java/org/jfrog/artifactory/client/model/repository/settings/impl/OciRepositorySettingsImpl.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        int result = super.hashCode();
+        return result;
+    }
```
Comment (yahavi) [inline services/src/main/java/org/jfrog/artifactory/client/model/repository/settings/impl/OciRepositorySettingsImpl.java:33]: There is no point in overriding this method
```suggestion
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Thanks for your contribution, @jcoste-orange!
Please consider my inline comments.

**Comment by jcoste-orange** [comment]
@yahavi I push a new commit to take into account your review.

**Review by yahavi — APPROVED** [review]
Thanks a lot for this, @jcoste-orange!

**Comment by yahavi** [comment]
@jcoste-orange 
Artifactory Java client version 2.18.0 has been released and is anticipated to be available on Maven Central within a few hours.
Please feel welcome to share your feedback here.

**Comment by jonesbusy** [comment]
Thanks! I think we miss also the 'helmoci' repo which should be also similar to the "generic" OCI type

**Comment by yahavi** [comment]
@jonesbusy   
Artifactory Java client version 1.19.0 has been released, featuring your [fix](https://github.com/jfrog/artifactory-client-java/pull/391) for this issue.  
Kindly let us know if that resolved the issue.

**Pull request merged**